tests/CatalogTests/Helpers/FeedHelpersTests.cs

```diff
@@ -321,11 +344,44 @@ public async Task DownloadMetadata2CatalogAsync_WithOnePackage_UpdatesStorage(bo
 
                 NupkgMetadata nupkgMetadata = GetNupkgMetadata("Newtonsoft.Json.9.0.2-beta1.nupkg");
 
+                PackageDeprecationItem deprecationItem;
+                switch (deprecationState)
+                {
+                    case PackageDeprecationItemState.NotDeprecated:
+                        deprecationItem = null;
+                        break;
+                    case PackageDeprecationItemState.Deprecated:
+                        deprecationItem = new PackageDeprecationItem(
+                            new[] { "first", "second" });
+                        break;
+                    case PackageDeprecationItemState.DeprecatedWithMessage:
+                        deprecationItem = new PackageDeprecationItem(
+                            new[] { "first", "second" },
+                            "this is the message");
+                        break;
+                    case PackageDeprecationItemState.DeprecatedWithAlternateRegistration:
+                        deprecationItem = new PackageDeprecationItem(
+                            new[] { "first", "second" }, 
+                            "this is the message", 
+                            "theId");
+                        break;
+                    case PackageDeprecationItemState.DeprecatedWithAlternateVersion:
+                        deprecationItem = new PackageDeprecationItem(
+                            new[] { "first", "second" },
+                            "this is the message",
+                            "theId", 
+                            "2.4.5");
+                        break;
+                    default:
+                        throw new ArgumentException(nameof(deprecationState));
+                }
```

Comment:
move to a separate method, also, there are some opportunities to reduce code duplication.